### PR TITLE
Add github action packaging workflow

### PR DESCRIPTION
**Pull request by cjh1**
Here we go! :crossed_fingers: it builds first time :smile:

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      # command works.
+      shell: bash -l {0}
+
+    - name: Set version environment variables used by CPack
+      working-directory: hexrdgui
+      run: |
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} full`
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} major`
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} minor`
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} patch`
```
Comment (psavery) [inline .github/workflows/main.yml:121]: I think these don't need to be "echo'ed". I did something similar [here](https://github.com/hydroframe/SandTank/blob/b1a6d50ba33187ab1fcb54c760ba71fa1cfb3ab4/.github/workflows/docker.yml#L50), where the script itself sets a variable, and I don't think an echo was needed when calling the script.

Comment (cjh1) [reply .github/workflows/main.yml:121]: But it's a python script?

Comment (psavery) [reply .github/workflows/main.yml:121]: I think github actions just checks all of the stdout that comes out of the `run` command, and if anything starts with `::set-env`, it considers it to be telling it to set a variable.

**Comment by psavery** [comment]
It works! :+1:

File: .github/workflows/main.yml (hunk 1 above)
Comment (psavery) [reply .github/workflows/main.yml:121]: Or `::set-output`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        shell: bash
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        path: hexrdgui
+
+    - name: Checkout HEXRD master for local build ( PRs only )
+      if: github.ref != 'refs/heads/master'
```
Comment (psavery) [inline .github/workflows/main.yml:43]: By the way, I think these could also be `if: github.event_name == 'pull_request'`. But it's probably not that important.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      shell: bash -l {0}
+
+    - name: Build the HEXRD conda package ( PRs only )
+      if: github.ref != 'refs/heads/master'
+      working-directory: hexrd
+      run: |
+          conda activate hexrd
+          mkdir output
+          conda build --output-folder output/ conda.recipe/
+          pwd
```
Comment (psavery) [inline .github/workflows/main.yml:88]: This might not be needed.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      with:
+        name: InstallOutput.log
+        path: ${{ github.workspace }}/hexrdgui/packaging/_CPack_Packages/*/InstallOutput.log
+
+    - name: Upload WIX log ( Windows only )
+      if: ${{ failure() && matrix.config.os == 'windows-latest'}}
+      uses: actions/upload-artifact@v2
+      with:
+        name: wix.log
+        path: ${{ github.workspace }}/hexrdgui/packaging/_CPack_Packages/WIX/wix.log
```
Comment (psavery) [inline .github/workflows/main.yml:147]: These are nice!

File: .github/workflows/main.yml (hunk 1 above)
Comment (cjh1) [reply .github/workflows/main.yml:121]: Ok, thanks, will fix that up

File: .github/workflows/main.yml (hunk 2 above)
Comment (cjh1) [reply .github/workflows/main.yml:43]: Yup, I think either work in this case

File: .github/workflows/main.yml (hunk 3 above)
Comment (cjh1) [reply .github/workflows/main.yml:88]: Removed

**Pull request merged**